> The error message suggests using `APP_ALLOW_MISSING_DEPS=true`, which would allow the build to continue even when certain dependencies are missing.

@coderabbit Added `APP_ALLOW_MISSING_DEPS`, and not getting a different error message: "error: 'nullptr' was not declared in this scope"

```
/home/runner/work/sima/sima/android-ndk-r16b/build/core/build-binary.mk:693: Android NDK: Module cocos2dcpp_shared depends on undefined modules: cocos_curl_static    
[armeabi] Compile++ thumb: cocos2dcpp_shared <= main.cpp
In file included from /home/runner/work/sima/sima/android-ndk-r16b/sources/cxx-stl/gnu-libstdc++/4.9/include/thread:35:0,
                 from /home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/3d/../base/CCDirector.h:[32](https://github.com/elimu-ai/sima/actions/runs/14971217757/job/42052380759#step:11:33),
                 from /home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/3d/../base/CCAsyncTaskPool.h:29,
                 from /home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/3d/../cocos2d.h:41,
                 from /home/runner/work/sima/sima/proj.android/jni/../../Classes/AppDelegate.h:4,
                 from /home/runner/work/sima/sima/proj.android/jni/hellocpp/main.cpp:1:
/home/runner/work/sima/sima/android-ndk-r16b/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \
  ^
In file included from /home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/3d/../base/CCDirector.h:[35](https://github.com/elimu-ai/sima/actions/runs/14971217757/job/42052380759#step:11:36):0,
                 from /home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/3d/../base/CCAsyncTaskPool.h:29,
                 from /home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/3d/../cocos2d.h:41,
                 from /home/runner/work/sima/sima/proj.android/jni/../../Classes/AppDelegate.h:4,
                 from /home/runner/work/sima/sima/proj.android/jni/hellocpp/main.cpp:1:
/home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/3d/../base/CCRef.h: In member function 'cocos2d::Ref* cocos2d::Clonable::copy() const':
/home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/3d/../base/CCRef.h:67:16: error: 'nullptr' was not declared in this scope
         return nullptr;
                ^
```

--- a/.github/workflows/android-build.yml
+++ b/.github/workflows/android-build.yml
@@ -96,17 +96,17 @@ jobs:
         working-directory: proj.android
         run: |
           export NDK_MODULE_PATH=$COCOS_X_ROOT/cocos2d-x-3.10:$COCOS_X_ROOT/cocos2d-x-3.10/cocos:$COCOS_X_ROOT/cocos2d-x-3.10/external
-
+          
           # Create jni/Application.mk to specify architecture
           echo "APP_ABI := armeabi" > jni/Application.mk
           echo "APP_STL := gnustl_static" >> jni/Application.mk
           echo "NDK_TOOLCHAIN_VERSION := 4.9" >> jni/Application.mk
           echo "APP_ALLOW_MISSING_DEPS := true" >> jni/Application.mk
-          echo "APP_CPPFLAGS := -std=c++11" >> jni/Application.mk
-
+          echo "APP_CPPFLAGS := -std=c++11 -frtti" >> jni/Application.mk
+          
           # Debug the environment
           echo "NDK_MODULE_PATH=$NDK_MODULE_PATH"
-
+          
           # Build with explicit ABI specification
           $ANDROID_NDK_HOME/ndk-build -C jni APP_ABI=armeabi
 

--- a/.github/workflows/android-build.yml
+++ b/.github/workflows/android-build.yml
@@ -102,6 +102,7 @@ jobs:
           echo "APP_STL := gnustl_static" >> jni/Application.mk
           echo "NDK_TOOLCHAIN_VERSION := 4.9" >> jni/Application.mk
           echo "APP_ALLOW_MISSING_DEPS := true" >> jni/Application.mk
+          echo "APP_CPPFLAGS := -std=c++11" >> jni/Application.mk
 
           # Debug the environment
           echo "NDK_MODULE_PATH=$NDK_MODULE_PATH"

--- a/Classes/RecorderScene/SL_MainEngine.cpp
+++ b/Classes/RecorderScene/SL_MainEngine.cpp
@@ -7,6 +7,7 @@
 
 #include "SL_MainEngine.h"
 //#include "cocos2d.h"
+#include <cstdlib>
 
 /*
 * Only support up-sampling

